## PRE PROCESADO DE DATOS ##


In [650]:
import pandas as pd
df = pd.read_csv('steam.csv')
df

appid                        name release_date  english  \
0           10              Counter-Strike   2000-11-01        1   
1           20       Team Fortress Classic   1999-04-01        1   
2           30               Day of Defeat   2003-05-01        1   
3           40          Deathmatch Classic   2001-06-01        1   
4           50   Half-Life: Opposing Force   1999-11-01        1   
...        ...                         ...          ...      ...   
27070  1065230             Room of Pandora   2019-04-24        1   
27071  1065570                   Cyber Gun   2019-04-23        1   
27072  1065650            Super Star Blast   2019-04-24        1   
27073  1066700  New Yankee 7: Deer Hunters   2019-04-17        1   
27074  1069460                   Rune Lord   2019-04-24        1   

                developer             publisher          platforms  \
0                   Valve                 Valve  windows;mac;linux   
1                   Valve                 Valve  windows;mac;linux   
2                   Valve                 Valve  windows;mac;linux   
3                   Valve                 Valve  windows;mac;linux   
4        Gearbox Software                 Valve  windows;mac;linux   
...                   ...                   ...                ...   
27070         SHEN JIAWEI           SHEN JIAWEI            windows   
27071      Semyon Maximov      BekkerDev Studio            windows   
27072         EntwicklerX           EntwicklerX            windows   
27073  Yustas Game Studio  Alawar Entertainment        windows;mac   
27074    Adept Studios GD  Alawar Entertainment        windows;mac   

       required_age                                         categories  \
0                 0  Multi-player;Online Multi-Player;Local Multi-P...   
1                 0  Multi-player;Online Multi-Player;Local Multi-P...   
2                 0              Multi-player;Valve Anti-Cheat enabled   
3                 0  Multi-player;Online Multi-Player;Local Multi-P...   
4                 0  Single-player;Multi-player;Valve Anti-Cheat en...   
...             ...                                                ...   
27070             0                   Single-player;Steam Achievements   
27071             0                                      Single-player   
27072             0  Single-player;Multi-player;Co-op;Shared/Split ...   
27073             0                          Single-player;Steam Cloud   
27074             0                          Single-player;Steam Cloud   

                       genres                 steamspy_tags  achievements  \
0                      Action        Action;FPS;Multiplayer             0   
1                      Action        Action;FPS;Multiplayer             0   
2                      Action  FPS;World War II;Multiplayer             0   
3                      Action        Action;FPS;Multiplayer             0   
4                      Action             FPS;Action;Sci-fi             0   
...                       ...                           ...           ...   
27070  Adventure;Casual;Indie        Adventure;Indie;Casual             7   
27071  Action;Adventure;Indie        Action;Indie;Adventure             0   
27072     Action;Casual;Indie           Action;Indie;Casual            24   
27073  Adventure;Casual;Indie        Indie;Casual;Adventure             0   
27074  Adventure;Casual;Indie        Indie;Casual;Adventure             0   

       positive_ratings  negative_ratings  average_playtime  median_playtime  \
0                124534              3339             17612              317   
1                  3318               633               277               62   
2                  3416               398               187               34   
3                  1273               267               258              184   
4                  5250               288               624              415   
...                 ...               ...               .

Este dataset muestra 27075 juegos publicados en steam. Cada fila obviamente pertenece a cada juego en particular . Luego contamos con varias columnas
que permite visualizar ciertas caracteristicas de cada juego publicado en steam

appid : Es un identificador unico  de un juego en steam. Uno de sus usos podria ser para dirigirnos a la pagina de tienda de un juego en particular

Ejemplo : El counter Strike tiene el identificador 10. Supongamos que nos interesa comprar ese juego, deberiamos ir a la pagina de steam y ver toda la informacion de ese juego con la opcion de comprar

https://store.steampowered.com/app/10/  

Name : Nombre del juego
Release Date : Fecha de lanzamiento del juego
English : ¿Si tiene el idioma ingles por defecto?
Developer : Nombre de la empresa que desarrollo el videojuego
platforms : Sistema operativos compatibles con el juego
required_age : ¿edad requerida para que una persona pueda comprar un juego?
categories : permite dar alguna informacion sobre si ese juego es en linea (multiplayer) o no (singleplayer) o ambos
Genero : Como pasa en los libros ,los juegos tambien suelen ser de un genero , como de accion ,terror , plataformas etc
steamspy_tags : Es como una mezcla entre categoria y genero segun podemos analizar. Deberia ser util para encontrar juegos que cumplan con nuestros intereses
achievements: Son logros. Se puede entender como que son desafios que ofrece el juego para  que un usuario pueda realizarlos si quiere. 
Su utilidad esta mas enfocado a lo social ,osea que un Usuario pueda compartir sus logros de un juego con otro y compararse.

positive_ratings: reseñas positivas que dejaron los usuarios que compraron o jugador un juego.
negative_ratings: reseñas negativas
average playtime : Steam tiene un registro del tiempo que cada jugador dedica a cada juego en la plataforma. Esta columna saca el promedio de tiempo de todos los jugadores que jugaron a un juego determinado
owners : Cantidad de jugadores que son propietarios del juego.
price : precio actual del juego en dolares




Comienzo del analisis...

Este dataset es una maravilla para realizar un analisis de datos en mi humilde opinion. Contamos con varias columnas candidatas a ser las variables dependientes. 
Los cuales son 

average_playtime
median_playtime
owners
price

Para el primer analisis vamosa intentar predecir los owners de cada juego segun el resto de columnas salvo ciertas columnas que yo de antemano se que no van aportar nada.

Pero antes de descartar las columnnas vamos a realizar unas pruebas

Existe valores nulos en el DF?

In [651]:
df.isnull().values.any()

False

Por suerte no existe los valores nulos en el dataframe. Eso no garantiza igualmente que el dato Nulo este representado de otra forma.

Pero antes de seguir profundizando con esto veo que a simple vista que en todos los juegos ,la columna english tiene como valor 1.
Si es  asi English es la primera columna candidata a ser eliminada.

In [652]:
(df['english'] == 1).all()

False

Dio falso. Entonces existe juegos en donde la columna english es diferente a 1. Comprobemoslo

In [653]:
df.loc[df['english'] != 1]

appid                                               name  \
1297    207370                       eXceed - Gun Bullet Children   
2221    262300                                        Tsukumogami   
3001    292990  NOBUNAGA'S AMBITION: Souzou (Traditional Chine...   
3085    295950                                             新流星搜劍錄   
3745    318300                   Phineas and Ferb: New Inventions   
...        ...                                                ...   
27021  1056010                                          Не падай!   
27024  1056500                                             VRLife   
27026  1056710                           VTB Basketball League VR   
27036  1058000                            Rain's love memory-雨的恋记   
27060  1062240                                           地铁：恐怖末班车   

      release_date  english                    developer  \
1297    2012-08-02        0                Flat Software   
2221    2012-05-31        0                      TORaIKI   
3001    2014-08-01        0   KOEI TECMO GAMES CO., LTD.   
3085    2017-12-20        0                       Titans   
3745    2015-02-24        0                  PIPE Studio   
...            ...      ...                          ...   
27021   2019-04-17        0                    SharF Co.   
27024   2019-04-11        0  Wuhan Mandrill VR Co., Ltd.   
27026   2019-04-12        0                  Axyos Games   
27036   2019-04-19        0             the original air   
27060   2019-04-25        0                   QZQ Studio   

                         publisher platforms  required_age  \
1297                     Nyu Media   windows             0   
2221              Fruitbat Factory   windows             0   
3001    KOEI TECMO GAMES CO., LTD.   windows             0   
3085                        Titans   windows            18   
3745            Disney Interactive   windows             0   
...                            ...       ...           ...   
27021                    SharF Co.   windows             0   
27024  Wuhan Mandrill VR Co., Ltd.   windows             0   
27026                       LikeVR   windows             0   
27036             the original air   windows             0   
27060                  H.P.Y.S,LLC   windows             0   

                                 categories  \
1297      Single-player;Steam Trading Cards   
2221                          Single-player   
3001              Single-player;Steam Cloud   
3085   Multi-player;Online Multi-Player;MMO   
3745                          Single-player   
...                                     ...   
27021      Single-player;Steam Achievements   
27024               Online Multi-Player;MMO   
27026                         Single-player   
27036                         Single-player   
27060                         Single-player   

                                                  genres  \
1297                                        Action;Indie   
2221                                 Adventure;Indie;RPG   
3001                                            Strategy   
3085   Violent;Action;Adventure;Indie;Massively Multi...   
3745                                           Adventure   
...                                                  ...   
27021                     Casual;Indie;Racing;Simulation   
27024  Casual;Free to Play;Massively Multiplayer;RPG;...   
27026                         Casual;Free to Play;Sports   
27036                                       Free to Play   
27060                         Action;Adventure;Indie;RPG   

                                         steamspy_tags  achievements  \
1297                    Bullet Hell;Anime;Shoot 'Em Up             0   
2221                               RPG;Indie;Adventure            49   
3001                               Strategy;Historical             0   
3085                               Action;RPG;Fighting             0   
3745     Adventure;Psychological Horror;Sexual Content             0

Antes de continuar , voy a verificar algo. <br>
<br>
¿sera que la columna  solo almacenas 1 y 0? ¿Sera una forma  de  representar los  booleanos?

In [654]:
valores_unicos = df['english'].unique()
valores_unicos

array([1, 0], dtype=int64)

Efectivamente es asi, la columna english solo almacena 0 y 1. Ahora volvamos analizar a los juegos con el campos english = 0

In [655]:
df.loc[df['english'] == 0]

appid                                               name  \
1297    207370                       eXceed - Gun Bullet Children   
2221    262300                                        Tsukumogami   
3001    292990  NOBUNAGA'S AMBITION: Souzou (Traditional Chine...   
3085    295950                                             新流星搜劍錄   
3745    318300                   Phineas and Ferb: New Inventions   
...        ...                                                ...   
27021  1056010                                          Не падай!   
27024  1056500                                             VRLife   
27026  1056710                           VTB Basketball League VR   
27036  1058000                            Rain's love memory-雨的恋记   
27060  1062240                                           地铁：恐怖末班车   

      release_date  english                    developer  \
1297    2012-08-02        0                Flat Software   
2221    2012-05-31        0                      TORaIKI   
3001    2014-08-01        0   KOEI TECMO GAMES CO., LTD.   
3085    2017-12-20        0                       Titans   
3745    2015-02-24        0                  PIPE Studio   
...            ...      ...                          ...   
27021   2019-04-17        0                    SharF Co.   
27024   2019-04-11        0  Wuhan Mandrill VR Co., Ltd.   
27026   2019-04-12        0                  Axyos Games   
27036   2019-04-19        0             the original air   
27060   2019-04-25        0                   QZQ Studio   

                         publisher platforms  required_age  \
1297                     Nyu Media   windows             0   
2221              Fruitbat Factory   windows             0   
3001    KOEI TECMO GAMES CO., LTD.   windows             0   
3085                        Titans   windows            18   
3745            Disney Interactive   windows             0   
...                            ...       ...           ...   
27021                    SharF Co.   windows             0   
27024  Wuhan Mandrill VR Co., Ltd.   windows             0   
27026                       LikeVR   windows             0   
27036             the original air   windows             0   
27060                  H.P.Y.S,LLC   windows             0   

                                 categories  \
1297      Single-player;Steam Trading Cards   
2221                          Single-player   
3001              Single-player;Steam Cloud   
3085   Multi-player;Online Multi-Player;MMO   
3745                          Single-player   
...                                     ...   
27021      Single-player;Steam Achievements   
27024               Online Multi-Player;MMO   
27026                         Single-player   
27036                         Single-player   
27060                         Single-player   

                                                  genres  \
1297                                        Action;Indie   
2221                                 Adventure;Indie;RPG   
3001                                            Strategy   
3085   Violent;Action;Adventure;Indie;Massively Multi...   
3745                                           Adventure   
...                                                  ...   
27021                     Casual;Indie;Racing;Simulation   
27024  Casual;Free to Play;Massively Multiplayer;RPG;...   
27026                         Casual;Free to Play;Sports   
27036                                       Free to Play   
27060                         Action;Adventure;Indie;RPG   

                                         steamspy_tags  achievements  \
1297                    Bullet Hell;Anime;Shoot 'Em Up             0   
2221                               RPG;Indie;Adventure            49   
3001                               Strategy;Historical             0   
3085                               Action;RPG;Fighting             0   
3745     Adventure;Psychological Horror;Sexual Content             0

el primer caso es eXceed - Gun Bullet Children. Tiene su appid 207370 asi que lo buscamos en la pagina de steam <br>

https://store.steampowered.com/app/207370/  

Efectivamente el juego no esta en ingles. Y al no estar en ingles puede ser determinante para que menos usuarios sean owners del juego en particular. Por lo que la columna english no se va a eliminar por ahora ya que a simple vista puede ser relevante para la prediccion. Sin embargo al entrar a la tienda del juego en steam vi que el numero de reseñas positivas del dataset no coincidian con las mostradas en la paginas... Antes de continuar veamos el juego

In [656]:

juegodeXceed = df.loc[df['appid'] ==207370, ['appid','name','english','genres', 'positive_ratings'	,'negative_ratings']]
juegodeXceed

appid                          name  english        genres  \
1297  207370  eXceed - Gun Bullet Children        0  Action;Indie   

      positive_ratings  negative_ratings  
1297               204               215

El juego eXceed - Gun Bullet Children que no esta en ingles cuenta con 204 reseñas positivas y 215 negativas en el dataset. En steam el numero es 227 y 261 respectivamente lo que ahora surge la duda (que en realidad no aporta nada a la prediccion) de que tan desactualizdo esta el dataset. Asi que veamos cual es el juego mas recientemente publicado en steam segun el dataset

In [657]:
fila_mas_reciente = df.loc[df['release_date'] == df['release_date'].max()]
fila_mas_reciente

appid               name release_date  english      developer  \
23726  905370  Conqueror's Blade   2019-05-01        1  Booming Games   

      publisher platforms  required_age  \
23726    My.com   windows             0   

                                              categories  \
23726  Online Multi-Player;MMO;Online Co-op;In-App Pu...   

                                                  genres        steamspy_tags  \
23726  Action;Free to Play;Massively Multiplayer;RPG;...  Action;Strategy;RPG   

       achievements  positive_ratings  negative_ratings  average_playtime  \
23726             0               259               235                 0   

       median_playtime   owners  price  
23726                0  0-20000  11.39

Con esto podriamos concluir que la ultima actualizacion del data set o la fecha misma de la creacion podria ser del 2019-05-01. Por lo que no vamos a encontrar en este dataset juego lanzados en 2023.

Entonces volviendo a lo que importa... La columna english no va a ser eliminada. <br>
Le toca a la columna required_age. Veamos que datos suele almacenar

In [658]:
df.loc[:,'required_age'].unique()

array([ 0, 18, 16, 12,  7,  3], dtype=int64)

Concluimos que existen juegos para personas de 18 años , otros para 16,12,7 y 3 inclusive. Esto podria ser determinante para la cantidad de owners que posee un juego, ya que en teoria se supone que esto excluye a ciertos jugadores que no cumple con la edad para jugar a un juego teoria. En la practica no suele ser tan asi ya que muchisimos menores de 18 años suelen jugar juegos para adultos. Aun asi existe la posibilidad de que algunos padres no dejen jugar a sus hijos a un juego si este determina que es un juego para adultos y jugador en cuestion no cuenta con  la edad suficiente.

Ahora como vemos arriba algunos juegos tienen 0 en la columna required_age ¿Que significado tiene esto? ¿Querra decir que incluso un recien nacido es apto para jugar el juego? ¿Lo que suena absurdo no? ¿O quizas querra decir que no se definio una edad , osea que es un null? <br>
Verifiquemoslo. Primero veamos cuantos juegos tenemos

In [659]:
num_filas = len(df)
num_filas

27075

ahora veamos cuantos juegos en la columna required_age tienen asignado el 0 como dato

In [660]:
len(df.loc[df['required_age'] == 0])

26479

Contamos con 26479 juegos que tienen almacenado 0 en la columna required_age.


In [661]:
porcentaje= (26479*100)/27075
porcentaje

97.79870729455217

¿Qué significa que el 98% de los juegos no pida ningún tipo de requisito de edad? ¿Es posible que los datos de required_age no se hayan almacenado correctamente y se interpreten los valores 0 como nulos? Verificar esto último es una tarea complicada, ya que requeriría buscar cada juego con required_age = 0 en la tienda de Steam y comprobar si realmente requiere una edad mínima para acceder. Si encontramos un solo juego con required_age = 0 que requiera una edad mínima en la aplicación web de Steam, concluiríamos que el valor 0 representa un valor nulo. En resumen, podemos asumir que el 98% de los juegos de Steam no requieren una edad mínima para acceder, aunque es posible que haya casos en los que la información de required_age no esté bien almacenada.

Para verificar que es cierto que los juegos de 18 años piden una edad minima (18 años por supuesto) hacemos una busqueda en el dataset primero

In [662]:
df.loc[df['required_age'] == 18]

appid                                name release_date  english  \
46        2210                            Quake IV   2011-08-04        1   
132       4560  Company of Heroes - Legacy Edition   2007-07-17        1   
136       4720         Condemned: Criminal Origins   2012-10-29        1   
173       6860                 Hitman: Blood Money   2007-03-15        1   
176       6900                 Hitman: Codename 47   2007-03-15        1   
...        ...                                 ...          ...      ...   
26325  1013180                      Funbag Fantasy   2019-01-31        1   
26372  1014730                               Cyndy   2019-01-31        1   
26444  1018810                       Drumpfy Walls   2019-04-19        1   
26499  1021860                       Unravel Cyndy   2019-02-10        1   
26695  1031920        Симулятор Сидения у Подъезда   2019-04-25        0   

                     developer           publisher platforms  required_age  \
46     RavenSoft / id Software  Bethesda Softworks   windows            18   
132        Relic Entertainment                SEGA   windows            18   
136                   Monolith                SEGA   windows            18   
173         Io-Interactive A/S  Io-Interactive A/S   windows            18   
176         IO Interactive A/S  IO Interactive A/S   windows            18   
...                        ...                 ...       ...           ...   
26325                   Waffle          MangaGamer   windows            18   
26372           DreamBig Games      DreamBig Games   windows            18   
26444            Hotbox Garage       Hotbox Garage   windows            18   
26499           DreamBig Games      DreamBig Games   windows            18   
26695        Kavkaz Sila Games   Kavkaz Sila Games   windows            18   

                                              categories  \
46                Single-player;Multi-player;Steam Cloud   
132    Single-player;Multi-player;Steam Trading Cards...   
136                                        Single-player   
173                                        Single-player   
176                                        Single-player   
...                                                  ...   
26325                          Single-player;Steam Cloud   
26372                                      Single-player   
26444  Single-player;Steam Achievements;Partial Contr...   
26499                                      Single-player   
26695                   Single-player;Steam Achievements   

                           genres                       steamspy_tags  \
46                         Action                  FPS;Action;Shooter   
132       Violent;Action;Strategy           Strategy;RTS;World War II   
136                        Action           Horror;Action;Atmospheric   
173                        Action         Stealth;Action;Third Person   
176                        Action              Stealth;Action;Classic   
...                           ...                                 ...   
26325                   Adventure     Sexual Content;Nudity;Adventure   
26372     Simulation;Early Access  Early Access;Nudity;Sexual Content   
26444  Action;Casual;Indie;Sports                 Action;Indie;Casual   
26499                       Indie         Nudity;Sexual Content;Indie   
26695                  Simulation                          Simulation   

       achievements  positive_ratings  negative_ratings  average_playtime  \
46                0               891               171                36   
132               0              4772               320               175   
136               0              1243               221               312   
173               0              5864               391               435   
176               0              1186               450                62   
...             ...               ...               ...               ...   
26325             0   

tomemos al primer juego. Quake 4 (Juego que curiosamente cuenta con una de las escena mas violenta de la historia de los videojuegos).
<br>
https://store.steampowered.com/app/2210/  


Como podemos ver ese juego nos pide que ingresemos nuestra edad...

En fin todo este analisis previo de datos se hizo con la intencion de profundizar un poco en el data set y ver la posibilidad de eliminar alguna columna de entrada . Al final no se elimino ninguna  ya que concluimos que no existen datos nulos camuflados en el 0 de las 2 columnas analizadas, aun asi  no se descarta eliminarlas junto otras nuevas   posteriomente de manera algoritmica.


Ahora se va intentar predecir el owner segun ciertas variables independientes pero el owner no es un dato numerico sino que intuyo que es categorico (Tiene un guion que representa un rango)


In [663]:
soloOwner = df.loc[:,['owners']]
soloOwner

owners
0      10000000-20000000
1       5000000-10000000
2       5000000-10000000
3       5000000-10000000
4       5000000-10000000
...                  ...
27070            0-20000
27071            0-20000
27072            0-20000
27073            0-20000
27074            0-20000

[27075 rows x 1 columns]

Lo que se va hacer ahora es convertir todos los datos de "categorico" a numerico .Para eso vamos a splitear cada  dato usando el guion como delimitador y luego los 2 valores numericos resultantes del corte se van a sumar y dividir por 2 para asi sacar un promedio y que ese promedio represente el la cantidad de owners que se hicieron con el juego.  Se crea una funcion para realizar esta tarea


In [664]:
def ownersPromedio(owner):
    nums = owner.split('-')
    num1 = int(nums[0])
    num2 = int(nums[1])
    return (num1 + num2) / 2

Ya tenemos la funcion. Ahora queremos aplicar la funcion a cada valor de la columna owner


In [665]:
df['ownersPromedio'] = df['owners'].apply(ownersPromedio)

visualizamos como va quedando la cosa

In [666]:
soloOwner = df.loc[:,['name','owners','ownersPromedio']]
soloOwner

name             owners  ownersPromedio
0                  Counter-Strike  10000000-20000000      15000000.0
1           Team Fortress Classic   5000000-10000000       7500000.0
2                   Day of Defeat   5000000-10000000       7500000.0
3              Deathmatch Classic   5000000-10000000       7500000.0
4       Half-Life: Opposing Force   5000000-10000000       7500000.0
...                           ...                ...             ...
27070             Room of Pandora            0-20000         10000.0
27071                   Cyber Gun            0-20000         10000.0
27072            Super Star Blast            0-20000         10000.0
27073  New Yankee 7: Deer Hunters            0-20000         10000.0
27074                   Rune Lord            0-20000         10000.0

[27075 rows x 3 columns]

Ahora se procede a eliminar la columna owners original

In [667]:
df = df.drop(columns=['owners'])

por ultimo la renombramos la columna ownersPromedio a owners

In [668]:
df = df.rename(columns={'ownersPromedio': 'owner'})

Listo . Visualicemos ahora el dataframe completo

In [669]:
df

appid                        name release_date  english  \
0           10              Counter-Strike   2000-11-01        1   
1           20       Team Fortress Classic   1999-04-01        1   
2           30               Day of Defeat   2003-05-01        1   
3           40          Deathmatch Classic   2001-06-01        1   
4           50   Half-Life: Opposing Force   1999-11-01        1   
...        ...                         ...          ...      ...   
27070  1065230             Room of Pandora   2019-04-24        1   
27071  1065570                   Cyber Gun   2019-04-23        1   
27072  1065650            Super Star Blast   2019-04-24        1   
27073  1066700  New Yankee 7: Deer Hunters   2019-04-17        1   
27074  1069460                   Rune Lord   2019-04-24        1   

                developer             publisher          platforms  \
0                   Valve                 Valve  windows;mac;linux   
1                   Valve                 Valve  windows;mac;linux   
2                   Valve                 Valve  windows;mac;linux   
3                   Valve                 Valve  windows;mac;linux   
4        Gearbox Software                 Valve  windows;mac;linux   
...                   ...                   ...                ...   
27070         SHEN JIAWEI           SHEN JIAWEI            windows   
27071      Semyon Maximov      BekkerDev Studio            windows   
27072         EntwicklerX           EntwicklerX            windows   
27073  Yustas Game Studio  Alawar Entertainment        windows;mac   
27074    Adept Studios GD  Alawar Entertainment        windows;mac   

       required_age                                         categories  \
0                 0  Multi-player;Online Multi-Player;Local Multi-P...   
1                 0  Multi-player;Online Multi-Player;Local Multi-P...   
2                 0              Multi-player;Valve Anti-Cheat enabled   
3                 0  Multi-player;Online Multi-Player;Local Multi-P...   
4                 0  Single-player;Multi-player;Valve Anti-Cheat en...   
...             ...                                                ...   
27070             0                   Single-player;Steam Achievements   
27071             0                                      Single-player   
27072             0  Single-player;Multi-player;Co-op;Shared/Split ...   
27073             0                          Single-player;Steam Cloud   
27074             0                          Single-player;Steam Cloud   

                       genres                 steamspy_tags  achievements  \
0                      Action        Action;FPS;Multiplayer             0   
1                      Action        Action;FPS;Multiplayer             0   
2                      Action  FPS;World War II;Multiplayer             0   
3                      Action        Action;FPS;Multiplayer             0   
4                      Action             FPS;Action;Sci-fi             0   
...                       ...                           ...           ...   
27070  Adventure;Casual;Indie        Adventure;Indie;Casual             7   
27071  Action;Adventure;Indie        Action;Indie;Adventure             0   
27072     Action;Casual;Indie           Action;Indie;Casual            24   
27073  Adventure;Casual;Indie        Indie;Casual;Adventure             0   
27074  Adventure;Casual;Indie        Indie;Casual;Adventure             0   

       positive_ratings  negative_ratings  average_playtime  median_playtime  \
0                124534              3339             17612              317   
1                  3318               633               277               62   
2                  3416               398               187               34   
3                  1273               267               258              184   
4                  5250               288               624              415   
...                 ...               ...               .